### PR TITLE
Issue 360 leak handler

### PR DESCRIPTION
**Pull request by dilandau2001**
Fixing memory leak on VlcMediaPlayer->GetMedia()

Adding a sample for multiple players tests. With this sample project I have intensively test the create and dispose of players so this #346 might be already fixed.

### CODE REVIEW

File: src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+        }
+
+        private void btnRemovePlayer_Click(object sender, EventArgs e)
+        {
+            this.RemovePlayer();
+        }
+
+        private void AddPlayer()
+        {
+            lock (listOfControls)
```
Comment (jeremyVignelles) [inline src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs]: I'm not a WinForms expert, but do you really need a lock there? I mean, you only call this method in button event handlers. Aren't these event triggered on the UI thread? That would make the lock unneeded.

Same below

File: src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+                // Dispose player
+                UnhookEvents(player);
+                player.Dispose();
+            }
+        }
+
+        private void button1_Click(object sender, EventArgs e)
+        {
+            lock (listOfControls)
```
Comment (jeremyVignelles) [inline src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs]: I'd replace the whole block with : `listOfControls.FirstOrDefault()?.Stop()`, but I'm ok with that.

File: src/Vlc.DotNet.Core/VlcMediaPlayer/VlcMediaPlayer.cs [hunk 3]
```diff
@@ -149,8 +152,8 @@ public void Play()
         /// <param name="options">The options to be given</param>
         public void Play(FileInfo file, params string[] options)
         {
-            this.SetMedia(file, options);
-            this.Play();
+            SetMedia(file, options);
```
Comment (jeremyVignelles) [inline src/Vlc.DotNet.Core/VlcMediaPlayer/VlcMediaPlayer.cs]: It's a cosmetic change. Can these be excluded from the PR?

File: src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs (hunk 1 above)
Comment (dilandau2001) [reply src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs]: I am planning to move the code to async/await so the form does not get totally frozen during player creation and deletion.
I have improved it a lot, although I found that player.EndInit is quite slow an must run on the ui thread.
You will see in the next commit

File: src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs (hunk 2 above)
Comment (dilandau2001) [reply src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs]: same as previous.

File: src/Vlc.DotNet.Core/VlcMediaPlayer/VlcMediaPlayer.cs (hunk 3 above)
Comment (dilandau2001) [reply src/Vlc.DotNet.Core/VlcMediaPlayer/VlcMediaPlayer.cs]: Ok. 
Would you prefer a different PR for cosmetics?
I kind of like the use of "this." but to keep consistency about the rest of the code I felt removing those were harmless.

File: src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs (hunk 1 above)
Comment (jeremyVignelles) [reply src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs]: Beware, the library is compatible with very old versions of .net (2.0 as far as I remember). Please don't break the compatibility by introducing async/await there

File: src/Vlc.DotNet.Core/VlcMediaPlayer/VlcMediaPlayer.cs (hunk 3 above)
Comment (jeremyVignelles) [reply src/Vlc.DotNet.Core/VlcMediaPlayer/VlcMediaPlayer.cs]: I don't think this is high priority. I also like `this.` but we don't have clear coding styles.

File: src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs (hunk 1 above)
Comment (dilandau2001) [reply src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs]: the library keeps being compatible. this change only affects the sample. 
Do you want me to revert the sample?

Comment (dilandau2001) [reply src/Samples/Samples.WinForms.MultiplePlayers/Form1.cs]: Well, you can't do this:

`

            lock (listOfControls)
            {
                await Task.Run(() =>
                {
                    //
                }
            }
`

but you can do this:

`

            await Task.Run(() =>
            {
                    lock(something)
            }
`

Anyway.

I updated the sample with the asyncronous code in the last commit.

File: src/Vlc.DotNet.Core/VlcMediaPlayer/VlcMediaPlayer.cs (hunk 3 above)
Comment (dilandau2001) [reply src/Vlc.DotNet.Core/VlcMediaPlayer/VlcMediaPlayer.cs]: It won't take much time to pass some automatic tools over the code to clean some cosmetic warnings, like:
- Using directive is not required by the code and can be safely removed
- Redundant use of "this.". Although we could argue about modern coding styles and personal preferences, but at least I would enforce the code to be consistent and having sometimes "this" and sometimes nothing is confusing.

Not sure about other warnings regarding C# 6.0 features and so on. But at least the first 2 I mention takes a few seconds to fix.

**Comment by dilandau2001** [comment]
All commits squashed in last commit:
[Commit](https://github.com/ZeBobo5/Vlc.DotNet/pull/477/commits/1af441bd4612c367a1754c0ee1d0bb934710e26e)

**Comment by dilandau2001** [comment]
Hi.
Everything done and comments applied.
Is there anything pending to complete the pull?

**Comment by jeremyVignelles** [comment]
Yes, time for me to review it :smile: I don't think I will have much time until monday

**Comment by jeremyVignelles** [comment]
Squashed and merged manually thanks!
I made a few changes to the sample projects used the libvlc nuget package and used a publicly available URL by default, so that anyone can test.

**Pull request closed without merging**